include/mercury/kes_dashboard_live_tiles_cli/ISerialisable.h

```diff
@@ -8,23 +8,21 @@
  */
 
 
-#ifndef __KES_DASHBOARD_LIVE_TILES_CLI_I_SERIALISABLE_H__
-#define __KES_DASHBOARD_LIVE_TILES_CLI_I_SERIALISABLE_H__
+#ifndef INCLUDE_MERCURY_KES_DASHBOARD_LIVE_TILES_CLI_ISERIALISABLE_H_
+#define INCLUDE_MERCURY_KES_DASHBOARD_LIVE_TILES_CLI_ISERIALISABLE_H_
 
 
-#include <memory>
-
 #include <parson.h>
 
 
 class ISerialisable {
 
-    public:
-        virtual ~ISerialisable() {}
+ public:  // Constructors & destructors.
+    virtual ~ISerialisable() {}
 
-    public:
-        virtual bool initialise(std::shared_ptr<JSON_Value> serialisedData) = 0;
-        virtual std::shared_ptr<JSON_Value> serialise() = 0;
+ public:  // Methods.
+    virtual bool initialise(JSON_Value* serialisedData) = 0;
+    virtual JSON_Value* serialise() const = 0;
```

Comment:
Why the regular pointer here?

Reply:
It would seem that if I construct a `shared_ptr<JSON_Array>` and then append `shared_ptr<JSON_Value>` to it, when the array gets cleaned it takes care of items itself. I was getting a double free through my tests. Parson is pretty good at not requiring you to clean up after yourself so wouldn't wrap many things in `shared_ptr`s.
https://github.com/kgabis/parson/blob/master/parson.h#L201